d2/src/main/java/com/linkedin/d2/balancer/properties/ClusterFailoutProperties.java

```diff
@@ -0,0 +1,83 @@
+package com.linkedin.d2.balancer.properties;
+
+import java.util.Map;
+import java.util.List;
+/*
+   Copyright (c) 2012 LinkedIn Corp.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+
+import java.util.stream.Collectors;
+import java.util.Collections;
+
+/**
+ * Configuration for a service's cluster level failout properties. These properties are used to control
+ * the flow of traffic between datacenters.
+ */
+public class ClusterFailoutProperties
+{
+  private final List<Map<String, Object>> _clusterFailoutRedirectConfigs;
+  private final List<Map<String, Object>> _clusterFailoutBucketConfigs;
```

Comment:
It'd be much better to have these as strongly typed objects instead of Maps (e.g. List<FailoutRedirectConfig> and List<FailoutBucketConfig>). You'll need to write more code in the serializer/deserializer, but the type safety and self documenting code is worth it. Or is this being done to keep LI specific properties out of the OS code? If that's the case where will these be consumed, and will there be a type safe Java object representing them?

Reply:
Hey Bryan - we're intentionally keeping this weakly typed, in order to avoid exposing LI concepts in rest.li. The strongly typed properties are provided in a class in a library that we use internally.

Reply:
Right, forgot about that decision. Thanks for the reminder.